From c131ebf976c1884fcf24cea8e413853dca989a51 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Sat, 2 Mar 2019 11:11:29 -0500
Subject: [PATCH] Don't check ConvertSigns boolean every sign save

property lookups arent super cheap. they synchronize, validate
and check security managers.

diff --git a/src/main/java/net/minecraft/server/TileEntitySign.java b/src/main/java/net/minecraft/server/TileEntitySign.java
index daf0aeb43f..91c99222d5 100644
--- a/src/main/java/net/minecraft/server/TileEntitySign.java
+++ b/src/main/java/net/minecraft/server/TileEntitySign.java
@@ -18,6 +18,7 @@ public class TileEntitySign extends TileEntity implements ICommandListener { //
     private static final boolean keepInvalidUnicode = Boolean.getBoolean("Paper.keepInvalidUnicode"); // Allow people to keep their bad unicode if they really want it
     private boolean privateUnicodeRemoved = false;
     public java.util.UUID signEditor;
+    private static final boolean CONVERT_LEGACY_SIGNS = Boolean.getBoolean("convertLegacySigns");
     // Paper end
 
     public TileEntitySign() {
@@ -36,7 +37,7 @@ public class TileEntitySign extends TileEntity implements ICommandListener { //
         }
 
         // CraftBukkit start
-        if (Boolean.getBoolean("convertLegacySigns")) {
+        if (CONVERT_LEGACY_SIGNS) { // Paper
             nbttagcompound.setBoolean("Bukkit.isConverted", true);
         }
         // CraftBukkit end
-- 
2.21.0

